src/read/pe/resource.rs

```diff
@@ -1,289 +1,195 @@
+            + header.number_of_named_entries.get(LE) as usize;
+        let entries = data
+            .read_slice::<pe::ImageResourceDirectoryEntry>(&mut offset, entries_count)
+            .read_error("Invalid resource table entries")?;
+        Ok(Self { header, entries })
     }
 }
 
-impl<'data> ResourceDirectoryEntry<'data> {
-    /// Returns true if the entry is a sub-directory
-    pub fn is_directory(&self) -> bool {
-        (self.entry.offset_to_data_or_directory.get(LE)
-            & crate::pe::IMAGE_RESOURCE_DATA_IS_DIRECTORY)
-            > 0
-    }
-
-    /// Returns the offset to the associated directory or data struct
-    pub fn data_offset(&self) -> u32 {
-        self.entry.offset_to_data_or_directory.get(LE) & 0x7FFF_FFFF
-    }
-
-    /// Returns true if the name is an custom string
-    pub fn has_string_name(&self) -> bool {
-        (self.entry.name_or_id.get(LE) & crate::pe::IMAGE_RESOURCE_NAME_IS_STRING) > 0
+impl pe::ImageResourceDirectoryEntry {
+    /// Returns true if the entry has a name, rather than an ID.
+    pub fn has_name(&self) -> bool {
+        self.name_or_id.get(LE) & pe::IMAGE_RESOURCE_NAME_IS_STRING != 0
     }
 
-    /// Returns the name string offset
+    /// Returns the section offset of the name.
     ///
-    /// Valid if `has_string_name()` returns true
-    fn name_offset(&self) -> u32 {
-        self.entry.name_or_id.get(LE) & 0x7FFF_FFFF
+    /// Valid if `has_name()` returns true.
+    fn name(&self) -> ResourceName {
+        let offset = self.name_or_id.get(LE) & !pe::IMAGE_RESOURCE_NAME_IS_STRING;
+        ResourceName { offset }
     }
 
-    /// Returns the name id
+    /// Returns the ID.
     ///
-    /// Valid if `has_string_name()` returns false
-    fn name_id(&self) -> u16 {
-        (self.entry.name_or_id.get(LE) & 0x0000_FFFF) as u16
+    /// Valid if `has_string_name()` returns false.
+    fn id(&self) -> u16 {
+        (self.name_or_id.get(LE) & 0x0000_FFFF) as u16
     }
 
     /// Returns the entry name
-    pub fn name(&self) -> ResourceNameOrId<'data> {
-        if self.has_string_name() {
-            ResourceNameOrId::Name(ResourceName {
-                directory_data: self.directory_data,
-                offset: self.name_offset(),
-            })
+    pub fn name_or_id(&self) -> ResourceNameOrId {
+        if self.has_name() {
+            ResourceNameOrId::Name(self.name())
         } else {
-            ResourceNameOrId::Id(self.name_id())
+            ResourceNameOrId::Id(self.id())
         }
     }
 
-    /// Returns the entry language code
-    ///
-    /// This is only valid a the level 2 of a standard resource directory structure.
-    ///
-    /// In a standard resource directory structure:
-    /// - level 0: entry.name_or_id is the resource type
-    /// - level 1: entry.name_or_id is the resource name
-    /// - level 2: entry.name_or_id is the language code
-    pub fn language_code(&self) -> Option<u16> {
```

Comment:
I'm fine with the removal, I was wondering if it was really useful. I can do with the name id.